cli/src/semgrep/commands/ci.py

```diff
@@ -223,14 +223,23 @@ def ci(
     state.error_handler.configure(suppress_errors)
     scan_handler = None
 
+    if allow_untrusted_validators and not run_secrets_flag:
+        logger.debug(
+            "Implicity enable secrets validation since allow_untrusted_validators is set."
+        )
+        run_secrets_flag = True
```

Comment:
Actually this seems like this is needlessly repetitive with the scan code. Do we really need both?

Reply:
The original intent was to ensure `semgrep ci` and `semgrep scan` to do the same thing when encountering `--allow_untrusted_validators`. I was toying with the idea of bringing this logic down to the common code that ci/scan use but that got messy really quickly. 

Also pad mentioned this in the [scope doc](https://www.notion.so/semgrep/Secrets-Semgrep-Scan-d277031960f647a18e4f49884b84bc14?d=e4aed080a6e948a6adc63f82cd56905c#fc574a5be3a040cb962cf5e39eda39e9) recently:
> [The idea is that semgrep scan will be just for semgrep OSS stuff, without any interaction with the cloud, and semgrep ci for all stuff cloud related and that requires pro feature (and being logged in)](https://www.notion.so/Secrets-Semgrep-Scan-d277031960f647a18e4f49884b84bc14?d=e4aed080a6e948a6adc63f82cd56905c&pvs=5#fc574a5be3a040cb962cf5e39eda39e9)

l'll drop the repetitive code on the `semgrep scan` side.

Reply:
Done in [0fdf442](https://github.com/semgrep/semgrep/pull/9382/commits/0fdf442f4b0ccd1fdd32a8be40ab8aed22afd9fe)